spec/job_template_spec.rb

```diff
@@ -31,4 +31,15 @@
       described_class.new(instance).launch(json)
     end
   end
+
+  describe '#survey_spec' do
+    let(:get) { instance_double("Faraday::Result", :body => instance.to_json) }
+
+    it "returns a survey spec if it exists" do
+      AnsibleTowerClient::Api.instance_variable_set(:@instance, api_connection)
+      survey = described_class.new(instance).survey_spec
+      expect(survey).to be_a Hash
+      expect(survey['related']['survey_spec']).to eq 'example.com/api'
```

Comment:
I think this 'related' is wrong.  I think your get is returning a job_template rather than the survey response.